spinalcordtoolbox/scripts/sct_dmri_denoise_patch2self.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+import sys
+import numpy as np
+import nibabel as nib
+from dipy.denoise.patch2self import patch2self
+
+from spinalcordtoolbox.image import add_suffix
+from spinalcordtoolbox.utils import SCTArgumentParser, Metavar, init_sct, printv, extract_fname, set_loglevel, list_type
+
+
+# DEFAULT PARAMETERS
+class Param:
+    # The constructor
+    def __init__(self):
+        self.debug = 0
+        self.verbose = 1  # verbose
+        self.remove_temp_files = 1
+        self.parameter = "ols"
+        self.file_to_denoise = ''
+        self.bval_file = ''
+        self.output_file_name = ''
+
+
+def get_parser():
+    parser = SCTArgumentParser(
+        description='Utility function to denoise diffusion MRI images. Return the denoised image and also the difference '
+                    'between the input and the output.The Patch2Self denoising algorithm is based on self-supervised denoising via statistical independence of noise, as described in the following publications:\n'
+                    '\n'
+                    '- Fadnavis et al. Patch2Self: Denoising Diffusion MRI with Self-supervised Learning. NeurIPS, 2020, Vol. 33. (https://arxiv.org/abs/2011.01355)\n'
+                    '- Schilling et al. Patch2Self denoising of diffusion MRI in the cervical spinal cord improves intra-cord contrast, '
+                    'signal modelling, repeatability, and feature conspicuity. Preprint, 2021. (https://doi.org/10.1101/2021.10.04.21264389)\n'
```

Comment:
Very sorry for the delay in getting to this PR! I've added a link to the preprint (as suggested by @jcohenadad), which looks like this on my end:

![Screenshot from 2022-01-24 12-16-59](https://user-images.githubusercontent.com/16181459/150831939-b6128abd-6a16-46ca-8876-0c34bdda62c2.png)